consensus/istanbul/backend/announce.go

```diff
@@ -145,35 +157,86 @@ func (sb *Backend) sendAnnounceMsgs() {
 	}
 }
 
-func (sb *Backend) sendIstAnnounce() error {
-	enode := sb.Enode()
-	if enode == nil {
+func (sb *Backend) generateIstAnnounce() ([]byte, error) {
+	block := sb.currentBlock()
+	selfEnode := sb.Enode()
+
+	if selfEnode == nil {
 		sb.logger.Error("Enode is nil in sendIstAnnounce")
-		return nil
+		return nil, nil
 	}
 
-	enodeUrl := enode.String()
+	enodeUrl := selfEnode.String()
 	view := sb.core.CurrentView()
+	incompleteEnodeUrl := enodeUrl[:strings.Index(enodeUrl, "@")]
+	ipData := enodeUrl[strings.Index(enodeUrl, "@"):]
+
+	regVals, err := sb.retrieveRegisteredValidators()
```

Comment:
nit: Since `regVals` is now containing both registered and active validators, perhaps rename this to regAndActiveVals.